truffle/com.oracle.truffle.object.dsl.processor/src/com/oracle/truffle/object/dsl/processor/layout/LayoutGenerator.java

```diff
@@ -0,0 +1,845 @@
+/*
+ * Copyright (c) 2015, 2016, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.truffle.object.dsl.processor.layout;
+
+import javax.lang.model.type.TypeKind;
+
+import com.oracle.truffle.object.dsl.processor.layout.model.LayoutModel;
+import com.oracle.truffle.object.dsl.processor.layout.model.NameUtils;
+import com.oracle.truffle.object.dsl.processor.layout.model.PropertyModel;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LayoutGenerator {
+
+    private static final boolean GENERATE_SHAPE_BOUNDARY_CONSTANT_CHECKS = Boolean.valueOf(
+                    System.getProperty("org.jruby.truffle.om.dsl.processor.layout.LayoutGenerator.GENERATE_SHAPE_BOUNDARY_CONSTANT_CHECKS", Boolean.FALSE.toString()));
+
```

Comment:
You may want to remove this JRuby like named constant. If you keep it, it has to be documented and tested.